homeassistant/components/homekit_controller/climate.py

```diff
@@ -170,8 +178,49 @@ def get_characteristic_types(self) -> list[str]:
             CharacteristicsTypes.TEMPERATURE_COOLING_THRESHOLD,
             CharacteristicsTypes.TEMPERATURE_HEATING_THRESHOLD,
             CharacteristicsTypes.SWING_MODE,
+            CharacteristicsTypes.ROTATION_SPEED,
         ]
 
+    @property
+    def fan_modes(self) -> list[str] | None:
+        """Return the available fan modes."""
+        if self.service.has(CharacteristicsTypes.ROTATION_SPEED):
+            return [FAN_OFF,FAN_LOW, FAN_MEDIUM, FAN_HIGH]
+        return None
+
+    @property
+    def fan_mode(self) -> str | None:
+        """Return the current fan mode."""
+        speed = self.service.value(CharacteristicsTypes.ROTATION_SPEED)
+        fan_mode = FAN_OFF
+       
+        #homekit seems to set value 0 33 66 100
+        if speed > ROTATION_SPEED_MEDIUM :
+            fan_mode = FAN_HIGH
+        elif speed > ROTATION_SPEED_LOW :
+            fan_mode = FAN_MEDIUM
+        elif speed > 0 :
+            fan_mode = FAN_LOW
+        elif speed <= 0 :    
+            fan_mode = FAN_OFF
+
+        return fan_mode
+
+    async def async_set_fan_mode(self, fan_mode: str) -> None:
+        rotation = 0
+        if fan_mode == FAN_LOW:
+            rotation = ROTATION_SPEED_LOW
+        elif fan_mode == FAN_MEDIUM:
+            rotation = ROTATION_SPEED_MEDIUM 
+        elif fan_mode == FAN_HIGH:
+            rotation = ROTATION_SPEED_HIGH 
+
+        _LOGGER.warning("set %d",rotation)
```

Comment:
This looks like debugging code that was left in. Can we get rid of it?

Reply:
removed , my fault (was for debug)